Create a FAQ to describe difference among notary, TUF, Notary Project specification, and notation (#328)
notaryproject/notaryproject.dev: content/en/docs/faq.md

**Q: How are Notary and Notary Project related?**

**A:** Notary, a tool for artifact signing based on The Update Framework (TUF), is developed by the Notary Project community and is hosted within the Notary Project's GitHub organization. So, notary is a tool, and Notary Project is the community that developed and maintains it.

**Q: What is the difference between Notary Project and TUF?**

**A:** The Notary Project, as a [GitHub organization](https://github.com/notaryproject), host multiple repositories including a [TUF](https://github.com/notaryproject/tuf) repository, which is a specific repository intended for prototyping the storage of TUF metadata in OCI-compliant registries.

**Q: What is Notary Project specification?**

**A:** Notary Project Specifications, mainly created between years 2021-2022, are a collection of specifications distributed across "subprojects" within the Notary Project. Notation CLI and libraries have implemented these specifications, which are also utilized by other OSS projects and/or vendor tools that seek to interoperate with the Notary Project tooling.

**Q: What is the difference between Notary and Notation?**

> **A:** Notary and Notation are both solutions for signing artifacts. The main difference between these two tools is that notary is based on [The Update Framework](https://theupdateframework.com) (TUF) specification while Notation is non-TUF-based. While notary has server and client components, Notation has a CLI and libraries. An example of a notary implementation is Docker Content Trust (DCT). In addition to signing artifacts, Notation handles artifact verification, signature portability, and integration with third-party key management solutions via a plugin model.

Reviewer: ```suggestion
**A:** Notary and Notation are both solutions for signing artifacts. The main difference between these two tools is that notary is based on [The Update Framework](https://theupdateframework.com) (TUF) specification while Notation is non-TUF-based. While notary has server and client components, Notation has a CLI and libraries. An example of a notary implementation is Docker Content Trust (DCT). In addition to signing artifacts, Notation handles artifact verification, signature portability, and integration with third-party key/certificate management solutions via a plugin model.
```

**A:** Ideally, we should validate the signing identity first and then use the public key in the signing identity to validate the artifact signature.
However, this will lead to poor performance in the case where the signature is not valid as there are lots of validations against the signing identity including network calls for revocations, and possibly we won't even need to read the trust store/trust policy if the signature validation fails.
Also, by validating artifact signature first we will still fail the validation if the signing identity is not trusted.

> ## [Notary Project Terms](https://notaryproject.dev/docs/glossary/)

Reviewer 2: Should we use a relative link for the glossary page?

Author: I tried the relative link but got an error during the process. I tried this here: `({{< relref "/docs/glossary" >}})`

Reviewer 2: Maybe you can remove the relative link in the sub-heading and add a sentence in line 80, such as 
`The frequently asked questions about Notary Project terms are listed below. See the [glossary]({{< ref "/docs/glossary" >}}) page to learn the detailed definition of each Notary Project term.`

Below are the frequently asked questions about Notary Project terms. For detailed definitions of each Notary Project term, please refer to the [glossary]({{< ref "/docs/glossary" >}}) page.

**Q: What is Notary Project?**

> **A:** The Notary Project is the [GitHub organization](https://github.com/notaryproject), the community, all specifications, and all repositories within the Notary Project, including community-released tools like [notary](https://github.com/notaryproject/notary) and [notation](https://github.com/notaryproject/notation).

Reviewer 3: I would recommend to change to:

> The Notary Project is a set of specifications and tools intended to provide a cross-industry standard for securing software supply chains by using authentic container images and other OCI artifacts. Notary Project is also the name of the GitHub organization that has multiple prominent subprojects like Notation, Notary Project Specification, and Notary. Very often we use the name Notary Project to refer to all the above as well as the community that drives the specifications and the implementations.

Reviewer 4: updated "Notary Project Specification" to "Notary Project specifications"

**Q: How are Notary and Notary Project related?**

> **A:** Notary, a tool for artifact signing based on The Update Framework (TUF), is developed by the Notary Project community and is hosted within the Notary Project's GitHub organization. So, notary is a tool, and Notary Project is the community that developed and maintains it.

Reviewer 4: ```suggestion
**A:** [Notary](https://github.com/notaryproject/notary), a tool for artifact signing based on The Update Framework (TUF), is developed by the Notary Project community and is hosted within the Notary Project's GitHub organization. So, notary is a tool, and Notary Project is the community that developed and maintains it.
```

**A:** [Notary](https://github.com/notaryproject/notary), a tool for artifact signing based on The Update Framework (TUF), is developed by the Notary Project community and is hosted within the Notary Project's GitHub organization. So, notary is a tool, and Notary Project is the community that developed and maintains it.

**Q: What is the difference between Notary Project and TUF?**

> **A:** The Notary Project, as a [GitHub organization](https://github.com/notaryproject), host multiple repositories including a [TUF](https://github.com/notaryproject/tuf) repository, which is a specific repository intended for prototyping the storage of TUF metadata in OCI-compliant registries.

Reviewer 3: I would suggest changing this to:

> [The Update Framework (aka TUF)](https://github.com/theupdateframework) is a CNCF graduated project that helps developers maintain the security of software update systems. TUF is a separate community and separate GitHub organization. One of the subprojects under Notary Project, [`notary`](https://github.com/notaryproject/notary), uses TUF for the implementation.

**Q: What is Notary Project specification?**

> **A:** Notary Project Specifications, mainly created between years 2021-2022, are a collection of specifications distributed across "subprojects" within the Notary Project. Notation CLI and libraries have implemented these specifications, which are also utilized by other OSS projects and/or vendor tools that seek to interoperate with the Notary Project tooling.

Reviewer 4: ```suggestion
**A:** [Notary Project Specifications](https://github.com/notaryproject/specifications), mainly created between years 2021-2022, are a collection of specifications distributed across "subprojects" within the Notary Project. Notation CLI and libraries have implemented these specifications, which are also utilized by other OSS projects and/or vendor tools that seek to interoperate with the Notary Project tooling.
```

> **A:** [Notary Project Specifications](https://github.com/notaryproject/specifications), mainly created between years 2021-2022, are a collection of specifications distributed across "subprojects" within the Notary Project. Notation CLI and libraries have implemented these specifications, which are also utilized by other OSS projects and/or vendor tools that seek to interoperate with the Notary Project tooling.

Reviewer 3: Due to some portability challenges with the TUF-based implementation in Notary, circa 2019, the Notary Project community decided to work on a portable signature specification. This resulted in the creation of the [specifications](https://github.com/notaryproject/specifications) subproject. The Notary Project specifications are shared across repositories under Notary Project as well as used by other open-source projects and/or vendor tools that want to interoperate with the Notary Project tooling. The Notary Project community plans to add other specifications in the future as our work on software supply chain evolves.

**Q: Does Notary Project signature specification leverage TUF?**

**A:** 

> The Notary Project [signature specification](https://github.com/notaryproject/specifications/blob/main/specs/signature-specification.md) does not leverage TUF. Instead, it leverages [COSE](https://github.com/notaryproject/specifications/blob/main/specs/signature-envelope-cose.md) and [JWS](https://github.com/notaryproject/specifications/blob/main/specs/) envelopes to produce and handle portable signatures. 

Reviewer 4: ```suggestion
No, the Notary Project [signature specification](https://github.com/notaryproject/specifications/blob/v1.0.0/specs/signature-specification.md) does not leverage TUF. The Notary Project signature specification supports [COSE](https://github.com/notaryproject/specifications/blob/v1.0.0/specs/signature-envelope-cose.md) and [JWS](https://github.com/notaryproject/specifications/blob/v1.0.0/specs/signature-envelope-jws.md) envelopes to produce and consume portable signatures using existing key management systems without the need of additional server infrastructure.
```

> **A:** No, the Notary Project [signature specification](https://github.com/notaryproject/specifications/blob/v1.0.0/specs/signature-specification.md) does not leverage TUF. The Notary Project signature specification supports [COSE](https://github.com/notaryproject/specifications/blob/v1.0.0/specs/signature-envelope-cose.md) and [JWS](https://github.com/notaryproject/specifications/blob/v1.0.0/specs/signature-envelope-jws.md) envelopes to produce and consume portable signatures using existing key management systems without the need of additional server infrastructure.

Reviewer 3: We do not need to go into details. I believe the first sentence suffices. Let's just keep this one:

> No, the Notary Project [signature specification](https://github.com/notaryproject/specifications/blob/v1.0.0/specs/signature-specification.md) does not leverage TUF.

**Q: What is the difference between Notary and Notation?**

> **A:** Notary and Notation are both solutions for signing artifacts. The main difference between these two tools is that notary is based on [The Update Framework](https://theupdateframework.com) (TUF) specification while Notation is non-TUF-based. While notary has server and client components, Notation has a CLI and libraries. An example of a notary implementation is Docker Content Trust (DCT). In addition to signing artifacts, Notation handles artifact verification, signature portability, and integration with third-party key/certificate management solutions via a plugin model.

Reviewer 2: ```suggestion
**A:** Notary and Notation are both solutions for signing artifacts. The main difference between these two tools is that notary is based on [The Update Framework](https://theupdateframework.com) (TUF) specification while Notation is the implementation of the new [Notary Project specifications](https://github.com/notaryproject/specifications/) which does not rely on TUF. While notary has server and client components, Notation has a CLI and libraries. An example of a notary implementation is Docker Content Trust (DCT). In addition to signing artifacts, Notation handles artifact verification, signature portability, and integration with third-party key/certificate management solutions via a plugin model.
```

> **A:** Notary and Notation are both solutions for signing artifacts. The main difference between these two tools is that notary is based on [The Update Framework](https://theupdateframework.com) (TUF) specification while Notation is non-TUF-based. While notary has server and client components, Notation has a CLI and libraries. An example of a notary implementation is Docker Content Trust (DCT). In addition to signing artifacts, Notation handles artifact verification, signature portability, and integration with third-party key/certificate management solutions via a plugin model.

Reviewer 4: ```suggestion
**A:** [Notary](https://github.com/notaryproject/notary) and [Notation](https://github.com/notaryproject/notation) are both solutions for signing artifacts. The main difference between these two tools is that notary is based on [The Update Framework](https://theupdateframework.com) (TUF) specification while Notation is non-TUF-based. While notary has server and client components, Notation has a CLI and libraries. An example of a notary implementation is Docker Content Trust (DCT). In addition to signing artifacts, Notation handles artifact verification, signature portability, and integration with third-party key/certificate management solutions via a plugin model.
```

Reviewer 2: @Roseline-Bassey For this paragraph, you can try to combine the suggested changes from @yizha1 and me, then polish it with your own words.

Author: Sure, thank you.

**Q: I've heard the term "Notary v2". What does this mean?**

> **A:** The term "notary v2" was previously used by the Notary Project and others. However, various meanings were ascribed to it, leading to its ambiguous usage with some people referring to it as the entire Notary Project and others as the Notation CLI. Because of this ambiguity, the term "notary v2" is no longer used by the Notary Project. While the term may still be visible in some articles on the internet, the name "notary v2" is no longer valid.

Reviewer 4: ```suggestion
**A:** The term "Notary v2" or "notary v2" was previously used by the Notary Project and others. However, various meanings were ascribed to it, leading to its ambiguous usage with some people referring to it as the entire Notary Project and others as the [Notation CLI](https://github.com/notaryproject/notation). Because of this ambiguity, the term "Notary v2" or "notary v2" is no longer used by the Notary Project. While the term may still be visible in some articles on the internet, the name "Notary v2" or "notary v2" is no longer valid.
```

> **A:**  [Notary](https://github.com/notaryproject/notary) and [Notation](https://github.com/notaryproject/notation) are both solutions for signing artifacts. The main difference between these tools is that notary is based on [The Update Framework](https://theupdateframework.com) (TUF) specification while Notation implements the new [Notary Project specifications](https://github.com/notaryproject/specifications/), which do not depend on TUF. While notary has server and client components, Notation has a CLI and libraries. An example of a notary implementation is Docker Content Trust (DCT). In addition to signing artifacts, Notation handles artifact verification, signature portability, and integration with third-party key/certificate management solutions via a plugin model.

Reviewer 3: Small nits here:

> [Notary](https://github.com/notaryproject/notary) and [Notation](https://github.com/notaryproject/notation) are both subprojects of Notary Project and offer capabilities for signing container images. Notary is based on [The Update Framework](https://theupdateframework.com) (TUF) and does not implement any of the Notary Project specifications while Notation implements the new [Notary Project signing specification and workflows](https://github.com/notaryproject/specifications/). While Notary has a server and client components, Notation has a CLI and libraries and leverages exsiting key management infrastructure and [OCI-compliant](https://opencontainers.org/) registries. An example of a notary implementation is Docker Content Trust (DCT). In addition to signing artifacts, Notation handles artifact verification, signature portability, and integration with third-party key/certificate management solutions via a plugin model.

Reviewer 5: I'd reverse the order of tools to `What is the difference between Notation and Notary?` and follow the same ordering in the answer.

> **A:** [Notation](https://github.com/notaryproject/notation) and [Notary](https://github.com/notaryproject/notary) are both subprojects of Notary Project and offer capabilities for signing container images. Notary is based on [The Update Framework](https://theupdateframework.com) (TUF) and does not implement any of the Notary Project specifications while Notation implements the new [Notary Project signing specification and workflows](https://github.com/notaryproject/specifications/). While Notary has a server and client components, Notation has a CLI and libraries and leverages existing key management infrastructure and [OCI-compliant](https://opencontainers.org/) registries. An example of a `notary` implementation is Docker Content Trust (DCT). In addition to signing artifacts, Notation handles artifact verification, signature portability, and integration with third-party key/certificate management solutions via a plugin model.

Reviewer 2: ```suggestion
**A:** [Notation](https://github.com/notaryproject/notation) and [Notary](https://github.com/notaryproject/notary) are both subprojects of Notary Project and offer capabilities for signing container images. Notary has a server and client components, Notation has a CLI and libraries, and leverages existing key management infrastructure and [OCI-compliant](https://opencontainers.org/) registries. An example of a `notary` implementation is Docker Content Trust (DCT). Notary is based on [The Update Framework](https://theupdateframework.com) (TUF) and does not implement any of the Notary Project specifications, while Notation implements the new [Notary Project signing specification and workflows](https://github.com/notaryproject/specifications/). In addition to signing artifacts, Notation handles artifact verification, signature portability, and integration with third-party key/certificate management solutions via a plugin model.
```

Reviewer 2: This suggestion comes from @gokarnm as mentioned in https://github.com/notaryproject/notaryproject.dev/pull/328#discussion_r1289487767

Reviewer 4: @FeynmanZhou You suggestion is till Notary comes the first, please take a look at the latest update. Thanks.

> **A:** The term "Notary v2" or "notary v2" was previously used by the Notary Project and others. However, various meanings were ascribed to it, leading to its ambiguous usage with some people referring to it as the entire Notary Project and others as the [Notation CLI](https://github.com/notaryproject/notation). Because of this ambiguity, the term "Notary v2" or "notary v2" is no longer used by the Notary Project. While the term may still be visible in some articles on the internet, the name "Notary v2" or "notary v2" is no longer valid.

Reviewer 3: I would suggest changing to:

> The term "Notary v2" or "notary v2" was previously used by members of the Notary Project community and others. However, various meanings were ascribed to it, leading to its ambiguous usage with some people referring to it as the entire Notary Project and others as the [Notation CLI](https://github.com/notaryproject/notation). Because of this ambiguity, the term "Notary v2" or "notary v2" is no longer used by the Notary Project community. While the term may still be visible in some articles on the internet, the name "Notary v2" or "notary v2" is only preserved for historical reasons and will not be used by the Notary Project community going forward.

**A:** Ideally, we should validate the signing identity first and then use the public key in the signing identity to validate the artifact signature.
However, this will lead to poor performance in the case where the signature is not valid as there are lots of validations against the signing identity including network calls for revocations, and possibly we won't even need to read the trust store/trust policy if the signature validation fails.
Also, by validating artifact signature first we will still fail the validation if the signing identity is not trusted.

## Notary Project Terms

> **Q: What is the difference between Notary, Notary v2, Notation, TUF, and Notary Project?**

Reviewer 3: I would split this into a couple of questions. Here my proposal:
- What is Notary Project?
- How are Notary and Notary Project related?
- What is the difference between Notary Project and TUF?
- What is Notary Project specification?
- Does Notary Project signature specification leverage TUF?
- What is the difference between Notary and Notation?
- I've heard the term "Notary v2". What does this mean?

Reviewer 6: This fine with me

Author: I have updated this PR to address Toddy's feedback except for this question: `Does Notary Project signature specification leverage TUF?` Could you please direct me to a document that could assist in clarifying the question?@toddysm  @zr-msft 
Thankk you.

Reviewer 2: > I've heard the term "Notary v2". What does this mean?

@toddysm @zr-msft  For the term `Notary v2`, it's worth stating that this term was previously used with different meanings in many places but will not be used in the Notary Project since it is unclear enough. 

Because this term was widely used by others and is still visible in some articles (on the internet). To avoid ambiguous usage, we remind users that the term `Notary v2` is not valid anymore. Does it make sense? cc @Roseline-Bassey

Reviewer 4: @Roseline-Bassey The answer for the following question is "No", and you can find what `Notary Project signature specification` is about from 4th paragraph of this PR https://github.com/notaryproject/.github/pull/32

```
Does Notary Project signature specification leverage TUF?
```

Author: Thanks @yizha1

Author: > > I've heard the term "Notary v2". What does this mean?
> 
> @toddysm @zr-msft For the term `Notary v2`, it's worth stating that this term was previously used with different meanings in many places but will not be used in the Notary Project since it is unclear enough.
> 
> Because this term was widely used by others and is still visible in some articles (on the internet). To avoid ambiguous usage, we remind users that the term `Notary v2` is not valid anymore. Does it make sense? cc @Roseline-Bassey

I think It'll be important to also mention some of the different interpretations that were/are associated with the term `Notary v2.` For example, For some, it represented the entirety of the project, and for others, it meant the notation CLI.  @FeynmanZhou

Author: > > I've heard the term "Notary v2". What does this mean?
> 
> @toddysm @zr-msft For the term `Notary v2`, it's worth stating that this term was previously used with different meanings in many places but will not be used in the Notary Project since it is unclear enough.
> 
> Because this term was widely used by others and is still visible in some articles (on the internet). To avoid ambiguous usage, we remind users that the term `Notary v2` is not valid anymore. Does it make sense? cc @Roseline-Bassey

I think It'll be important to also mention some of the different interpretations that were/are associated with the term `Notary v2.` For example, For some, it represented the entirety of the project, and for others, it meant the notation CLI.  @FeynmanZhou

> 

Reviewer 4: Should we point to the glossary page directly?

Author: Yes. `Notary Project Terms` can point to the glossary page directly.

> **Q: What is the difference between Notary, Notation, TUF, and Notary Project Specifications?**

Reviewer 2: ```suggestion
**Q: What is the difference between Notary, Notary v2, Notation, TUF, and Notary Project?**
```

> **A:** Notary and Notation are both artifact signing solutions, but they differ in certain key aspects. Notary is based on [The Update Framework](https://github.com/theupdateframework/specification) (TUF) specification and has both server and client components for interacting with trusted collections. Notation, on the other hand, is not based on TUF and has a client component and libraries. Notation is responsible for signing, verification, and signature portability, as well as integration with third-party key management solutions via a plugin model. The Update Framework (TUF) is a model for ensuring security in package management. Notary is based on this model. However, TUF, as a project under the Notary Project, aims to implement the full TUF specification in a registry-native way, which is an improvement over Notary. Meanwhile, Notary Project Specifications describe how other open source projects and/or vendor tools can interoperate with the Notary project tooling. For further details on each term, please refer to the glossary page.

Reviewer 2: This paragraph is not very accurate. Could you borrow language from https://github.com/notaryproject/notaryproject.dev/pull/326/files and https://github.com/notaryproject/.github/issues/35?

For `Notary v2`, we can state that this term was previously used with different meanings but will not be used in the Notary Project since this term is unclear enough.

Author: I have updated the paragraph, please kindly review it again.
